src/app/core/services/github.service.ts

```diff
@@ -495,32 +495,55 @@ export class GithubService {
   }
 
   /**
-   * Returns an async function that will accept a GraphQL query that requests for paginated items.
-   * Said function will recursively query for all subsequent pages until a page that has less than 100 items is found,
-   * then return all queried pages in an array.
+   * Returns an observable that will continually emit the currently accumulated results, until a page that has less
+   * than 100 items is found, after which it performs a final emit with the full results array, and completes.
    *
    * @callback pluckEdges - A function that returns a list of edges in a ApolloQueryResult.
+   * @params query - The query to be performed.
+   * @params variables - The variables for the query.
    * @returns an async function that accepts a GraphQL query for paginated data and any additional variables to that query
    */
-  private withPagination<T>(pluckEdges: (results: ApolloQueryResult<T>) => Array<any>) {
-    return async (query: DocumentNode, variables: { [key: string]: any } = {}): Promise<Array<ApolloQueryResult<T>>> => {
-      const maxResultsCount = 100;
-      const cursor = variables.cursor || null;
-      const graphqlQuery = this.apollo.watchQuery<T>({ query, variables: { ...variables, cursor } });
-      return graphqlQuery.refetch().then(async (results: ApolloQueryResult<T>) => {
+  private withPagination<T>(
+    pluckEdges: (results: ApolloQueryResult<T>) => Array<any>,
+    query: DocumentNode,
+    variables: { [key: string]: any } = {}
+  ): Observable<ApolloQueryResult<T>[]> {
+    const maxResultsCount = 100;
+    const apollo = this.apollo;
+
+    let accumulatedResults: ApolloQueryResult<T>[] = [];
+    const behaviorSubject: BehaviorSubject<ApolloQueryResult<T>[]> = new BehaviorSubject(accumulatedResults);
+    const observable = behaviorSubject.asObservable();
```

Comment:
```suggestion
```
This variable is not used throughout the function. I think we can just do 
```typescript
return behaviorSubject.asObservable();
```
to make te method less cluttered.